### PR TITLE
Changes To Support A CPP API Wrapper

### PR DESCRIPTION
**Pull request by slacrherbst**
Changes to support a CPP API Wrapper. Also remove root level set/get functions.

### CODE REVIEW

File: python/pyrogue/_Root.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def decrement(self):
 
     def _check(self):
         if len(self._list) != 0 and (self._count == 0 or (self._period != 0 and (time.time() - self._last) > self._period)):
-            #print(f"Update fired {time.time()}")
+            #print(f"Utpdate fired {time.time()}")
```
Comment (ruck314) [inline python/pyrogue/_Root.py:66]: @slacrherbst Need to fix this typo

File: tests/api_test/src/api_test.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#include <rogue/interfaces/api/Bsp.h>
```
Comment (ruck314) [inline tests/api_test/src/api_test.cpp:1]: @slacrherbst Missing licensing code header

**Comment by ruck314** [comment]
@slacrherbst Are you going to add any `.rst` file to provide a HowTo for the users? Or later pull request?

File: python/pyrogue/_Root.py (hunk 1 above)
Comment (ruck314) [reply python/pyrogue/_Root.py]: @slacrherbst Did you accidently delete `self._testFunc` in that commit while you were fixing the typo?

https://github.com/slaclab/rogue/pull/945/commits/e0176f3e94b1c96bc4ef9a7ea39aa39bfef2f45f

**Comment by slacrherbst** [comment]
> @slacrherbst Are you going to add any `.rst` file to provide a HowTo for the users? Or later pull request?

We will add that in the full release later.

**Review by bengineerd — APPROVED** [review]

**Pull request merged**